Sources/ValidatingProperty.swift

```diff
@@ -0,0 +1,307 @@
+		_ inner: MutableProperty<Value>,
+		with other: Other,
+		_ validator: @escaping (Value, Other.Value) -> ValidatorOutput<Value, ValidationError>
+	) {
+		let other = Property(other)
+
+		self.init(inner) { input in
+			return validator(input, other.value)
+		}
+
+		other.signal
+			.take(during: lifetime)
+			.observeValues { [weak self] _ in
+				if let s = self {
+					switch s.result.value {
+					case let .failure(failedValue, _):
+						s.value = failedValue
+					case let .substitution(substitutedValue, _, _):
+						s.value = substitutedValue
+					case let .success(value):
+						s.value = value
+					}
+				}
+		}
+	}
+
+	/// Create a `MutableValidatingProperty` that validates mutations before
+	/// committing them.
+	///
+	/// The proposed value is only committed when `success` is returned by the
+	/// `validator` closure.
+	///
+	/// - parameters:
+	///   - initial: The initial value of the property. It is not required to
+	///              pass the validation as specified by `validator`.
+	///   - other: The property that `validator` depends on.
+	///   - validator: The closure to invoke for any proposed value to `self`.
+	public convenience init<Other: PropertyProtocol>(
+		_ initial: Value,
+		with other: Other,
+		_ validator: @escaping (Value, Other.Value) -> ValidatorOutput<Value, ValidationError>
+	) {
+		self.init(MutableProperty(initial), with: other, validator)
+	}
+
+	/// Create a `MutableValidatingProperty` that presents a mutable validating
+	/// view for an inner mutable property.
+	///
+	/// The proposed value is only committed when `success` is returned by the
+	/// `validator` closure.
+	///
+	/// - note: `inner` is retained by the created property.
+	///
+	/// - parameters:
+	///   - inner: The inner property which validated values are committed to.
+	///   - other: The property that `validator` depends on.
+	///   - validator: The closure to invoke for any proposed value to `self`.
+	public convenience init<U, E: Swift.Error>(
+		_ inner: MutableProperty<Value>,
+		with other: MutableValidatingProperty<U, E>,
+		_ validator: @escaping (Value, U) -> ValidatorOutput<Value, ValidationError>
+	) {
+		self.init(inner, with: other, validator)
+	}
+
+	/// Create a `MutableValidatingProperty` that validates mutations before
+	/// committing them.
+	///
+	/// The proposed value is only committed when `success` is returned by the
+	/// `validator` closure.
+	///
+	/// - parameters:
+	///   - initial: The initial value of the property. It is not required to
+	///              pass the validation as specified by `validator`.
+	///   - other: The property that `validator` depends on.
+	///   - validator: The closure to invoke for any proposed value to `self`.
+	public convenience init<U, E: Swift.Error>(
+		_ initial: Value,
+		with other: MutableValidatingProperty<U, E>,
+		_ validator: @escaping (Value, U) -> ValidatorOutput<Value, ValidationError>
+	) {
+		let otherValidations = other.result
```

Comment:
Is there a reason that this is outside the `init`? If so, can you add a comment explaining why?